src/state/routing/types.ts

```diff
@@ -105,3 +104,43 @@ export class InterfaceTrade<
     this.gasUseEstimateUSD = gasUseEstimateUSD
   }
 }
+
+export type InterfaceTrade = ClassicTrade<Currency, Currency, TradeType>
+
+export enum QuoteState {
+  SUCCESS = 'Success',
+  INITIALIZED = 'Initialized',
+  NOT_FOUND = 'Not found',
+}
+
+export type QuoteResult =
+  | {
+      state: QuoteState.INITIALIZED | QuoteState.NOT_FOUND
+      data?: undefined
```

Comment:
nit: `data: undefined`

Reply:
this one you actually need to have `data?: undefined` here or else it will explicitly require that consumers pass in `{ state: whatever, data: undefined }`